### PR TITLE
Optimize Sender Recovery Process

### PR DESCRIPTION
**Pull request by TheDhejavu**
An attempt to resolve this: https://github.com/paradigmxyz/reth/issues/10835

## Description 
Spawn a single thread to consume chunks from each batch. The **Producer** prepares and sends batch chunks, while the **Consumer** processes these chunks using global Rayon workers. Finally, the producer collects and processes the results, managing both successful completions and any potential errors.

```mermaid
flowchart TD
    subgraph SenderRecoveryStage.execute
        subgraph Producer
            direction TB
            A[Start] --> B[Create Chunks from Batches]
            B --> C[Send Chunks to Channel]
       
            K[Receive Results from Consumer]
            K --> |Success| L[Process Success]
            K --> |Error| M[Handle Error]
            L --> N[End]
            M --> N[End]
        end
    end

    subgraph Consumer["Single thread"]
        direction TB
        E[Start] --> F[Receive Chunks]
        F --> G[Process Chunks with Rayon]
        G --> H[Send Results to Producer]
        H --> I[End]
    end

    C --> |Chunks| F
    H --> |Results| K

    classDef default fill:#dc14,stroke:#000,stroke-width:1px;


```

### CODE REVIEW

**Comment by TheDhejavu** [comment]
Failing tests. working on fixing that.

**Comment by TheDhejavu** [comment]
cc @joshieDo @mattsse

File: crates/stages/stages/src/stages/sender_recovery.rs [hunk 1]
```diff
@@ -267,10 +220,79 @@ where
             .into(),
         ));
     }
-
     Ok(())
 }
 
+fn setup_range_recovery<Provider>(
```
Comment (mattsse) [inline crates/stages/stages/src/stages/sender_recovery.rs:226]: can we add some docs here, ideally this fn name should mention spawn

**Review by mattsse — CHANGES_REQUESTED** [review]
cool, this is exactly what we had in mind, ty

some doc nits, and @joshieDo sanity check

**Review by mattsse — APPROVED** [review]
nice, tysm!

pending @joshieDo

**Review by joshieDo — APPROVED** [review]

**Comment by TheDhejavu** [comment]
is there anything left to do here cc @mattsse @joshieDo

**Comment by mattsse** [comment]
sorry forgot about to pull the trigger on this one

**Pull request merged**